packages/vats/src/provisionPoolKit.js

```diff
@@ -238,24 +242,26 @@ export const prepareProvisionPoolKit = (
           revivableAddresses.has(address) ||
             Fail`non-revivable address ${address}`;
           const bank = E(bankManager).getBankForAddress(address);
-          const [wallet, _created] = await E(walletFactory).provideSmartWallet(
+          return E(walletFactory).provideSmartWallet(
             address,
             bank,
             namesByAddressAdmin,
           );
-          return wallet;
         },
         /**
+         * @template T
          * @param {string} address
-         * @returns {boolean} isRevive
+         * @param {T} wallet
+         * @returns {[wallet: T, isNew: boolean]}
          */
-        ackWallet(address) {
+        ackWallet(address, wallet) {
           const { revivableAddresses } = this.state;
-          if (!revivableAddresses.has(address)) {
-            return false;
+          const isRevive = revivableAddresses.has(address);
+          if (isRevive) {
+            revivableAddresses.delete(address);
           }
-          revivableAddresses.delete(address);
-          return true;
+          const isNew = !isRevive;
```

Comment:
Why have two flag variables that encode the same thing but with opposite polarity?

Reply:
OMG these variables have the same name as types! Every time I've found myself in that situation, I've gotten very confused.